homeassistant/components/weheat/sensor.py

```diff
@@ -68,6 +145,9 @@ async def async_setup_entry(
         WeheatHeatPumpSensor(coordinator, entity_description)
         for entity_description in SENSORS
         for coordinator in entry.runtime_data
+        if SENSOR_DHW_KEY
+        not in entity_description.key  # only add Domestic Hot water (DHW) related sensors if the install has a DHW vessel
+        or coordinator.heat_pump_info.has_dhw
```

Comment:
Rather split out the DHW a separate list and just extend the entity list if the heat pump has dhw. Although this works, it's a strange way of noting (and the fact that it needs a comment to explain says that the code is not clear to start with)